### PR TITLE
Added support to make auto close select window optional

### LINKED ISSUE #137: Prevent the dropdown from closing when a result is selected
Please Add [this feature](https://select2.github.io/options.html#can-i-prevent-the-dropdown-from-closing-when-a-result-is-select) to this grate library.

### PR DESCRIPTION
**Pull request by yousharizvi**
Fix #137

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ var packages = {
 | notFoundMsg        | `string`          | `"No results found"`  | The message shown if no options are found for the current filter input value.              |
 | placeholder        | `string`          | `""`                  | Placeholder text that is shown if no options are selected.
 | filterPlaceholder  | `string`          | `""`                  | Placeholder text that is shown on the filter input (**single select only**).
+| autoClose          | `boolean`         | `true`                | If set to true, the select window won't close on selecting/deselecting options.            |
```
Comment (Crevil) [inline README.md]: Should this not be: If set to false ?

Comment (yousharizvi) [inline README.md]: Oh, that's a typo, pushing with correction now

**Comment by yousharizvi** [comment]
@basvandenberg, please see this. Also a total of 5 PRs are open.

**Comment by basvandenberg** [comment]
Thanks for your contribution! I very much like the `autoClose` option, although I would rather use the same name as select2, `closeOnSelect`, which in my opinion is a more accurate description of what the option does.

I fear that this feature requires a bit more work though. When `closeOnSelect` is set to false:
- Drop down should remain open when user clicks an option (this is implemented by this PR).
- Drop down should remain open when user selects an option by pressing the ENTER key.

There are also some other difficulties:
- In case of multiselect, if the number of selected options do not fit on one line anymore, the select container grows. In that case the dropdown list should be moved down to underneath the select container, which currently does not happen. 
- If one of the options in the drop down is clicked, the key handlers do not work any more.
- The same holds for deselecting options in the multiselect case. If you click the little cross of one of the selected options, focus changes and key handling is totally messed up.

I don't think this is very straightforward get this to work. I will have a look at it later.

**Comment by basvandenberg** [comment]
The repository is not being maintained and will be archived soon.

**Pull request closed without merging**